### PR TITLE
Add implementation of screen "History of operations"

### PR DESCRIPTION
**Pull request by levvaa**
### What

Added implementation of screen "History of operations"

### Why

It's part of the work on issue [Create UI for History of Operations screen](https://github.com/ios-course/link-team-project/issues/3)

### How

- Added file for History of Operations
- Added main title of the screen
- Added button "Add". 
- Added BillsListView
- Added ModelBill file. This file contains temporary model, that will be changed in future.

This PR does not include the implementation of a blank screen displaying the message "History of operations list is empty. Click "Add" to add a new bill"

### CODE REVIEW

File: JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import SwiftUI
+
+/// A view that displays a list of bills grouped by date.
```
Comment (kirpichonok) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift:3]: /// Displays a list of bills grouped by date.

Comment (kirpichonok) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift:3]:
```suggestion
/// Displays a list of bills grouped by date.
```

File: JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+struct BillListView: View {
+    var bills: BillsModel = .init()
```
Comment (kirpichonok) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift:5]:
```suggestion
    var bills =  BillsModel ()
```
I suggest to use simpler syntax.

File: JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    /// A computed property that groups the bills by date.
```
Comment (kirpichonok) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift:7]:
```suggestion
    /// Shows the bills grouped by date.
```

File: JourneysSquad/JourneysSquad/HistoryOfOperations/HistoryOfOperationsView.swift [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import SwiftUI
+
+/// A view that displays the history of operations.
+struct HistoryOfOperationsView: View {
+    var body: some View {
+        VStack {
+            Button(action: {}) { // TODO: Add action.
```
Comment (kirpichonok) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/HistoryOfOperationsView.swift:7]: Why do you need `// TODO: ...` here? 
Is it your incomplete task?

Comment (kirpichonok) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/HistoryOfOperationsView.swift:7]: Why didn't  you use navigation bar settings to add `Title` and `Button`?

File: JourneysSquad/JourneysSquad/HistoryOfOperations/HistoryOfOperationsView.swift [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                .padding(.top, 5)
+
+            Divider()
+                .background(Color.black)
+
+            Spacer()
+
+            BillListView()
+        }
+        .padding(.horizontal, 15)
```
Comment (kirpichonok) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/HistoryOfOperationsView.swift:25]: I think that `.padding(.horizontal, 5)` is enough.

File: JourneysSquad/JourneysSquad/HistoryOfOperations/BillsModel.swift [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+/// Describes data, that contains in bill.
+struct BillsModel {
```
Comment (kirpichonok) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/BillsModel.swift:4]: The name  `DummyBillModel` makes its purpose cleaner.

**Review by kirpichonok — CHANGES_REQUESTED** [review]
- I suggest to create a folder View and add to that `view` files. You can add a folder "History of Operations" inside folder "View". 
- Bring your code in line with our formatting rules.
- Update CHANGELOG.md
- Update Target's and Project's versions.

File: JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    var groupedBills: [String: [[String: Any]]] {
```
Comment (dersim-davaod) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift:8]: So each time you access the `groupedBills` the operation of grouping is performed. What is the complexity of such operation?

File: JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        Dictionary(grouping: bills.items) { bill in
+            (bill["date"] as? String) ?? ""
```
Comment (dersim-davaod) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift:10]: Magic string.

File: JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+
+    var body: some View {
+        List {
+            // Iterate over each group of bills, sorted by date in descending order.
```
Comment (dersim-davaod) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift:16]: Please remove comments.
If you do believe the comment should be here, that probably means your code is not readable.

File: JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            ForEach(groupedBills.sorted(by: { $0.key > $1.key }), id: \.key) { date, bills in
```
Comment (dersim-davaod) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift:17]: Shouldn't we prepare all the data so the view just displays it (that means no additional operations/sorting/etc)?

Could you please think of another view model?

Comment (dersim-davaod) [inline JourneysSquad/JourneysSquad/HistoryOfOperations/BillListView.swift:17]: The implementation is hard to read.

First, please extract the section and the row into a separate view.

**Review by dersim-davaod — CHANGES_REQUESTED** [review]
I see a lot of warnings. Please fix them.
Also please format the code before pushing the changes to the remote.
Additionally, please consider the following improvements:

**Comment by levvaa** [comment]
the current pull request will be split into subtasks with separate pull requests

**Comment by levvaa** [comment]
Issue has been split into smaller issues #44 #45 #42 #43 with separate pull requests.

**Pull request closed without merging**